### PR TITLE
Collect more data when deploy job fails to sync time

### PR DESCRIPTION
**Pull request by citrus-it**

### CODE REVIEW

File: .github/buildomat/jobs/deploy.sh [hunk 1]
```diff
@@ -65,12 +66,16 @@ _exit_trap() {
...
 	for p in $PORTS; do
+		pfexec /opt/oxide/opte/bin/opteadm dump-uft -p $p
 		LAYERS=$(pfexec /opt/oxide/opte/bin/opteadm list-layers -p $p | tail +2 | awk '{ print $1; }')
 		for l in $LAYERS; do
 			pfexec /opt/oxide/opte/bin/opteadm dump-layer -p $p $l
 		done
 	done
 
+	mkdir -p /tmp/kstat
+	pfexec kstat -p xde: > /tmp/kstat/xde.kstat
```
Comment (FelixMcFelix) [inline .github/buildomat/jobs/deploy.sh:77]: Had a quick look at the failed CI run there, this gives us all the pretty much numbers we can get on every OPTE layer. Thanks!

File: docs/debugging-time-sync.adoc [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+----
+root@oxz_ntp_cb901d3e:~# svcadm disable ntp
+root@oxz_ntp_cb901d3e:~# /usr/sbin/chronyd -t 10 -ddQ 'pool time.cloudflare.com iburst maxdelay 0.1'
+2024-10-02T17:02:54Z chronyd version 4.5 starting (+CMDMON +NTP +REFCLOCK -RTC
++PRIVDROP -SCFILTER +SIGND +ASYNCDNS -NTS +SECHASH +IPV6 -DEBUG)
+2024-10-02T17:02:54Z Disabled control of system clock
+2024-10-02T17:02:59Z System clock wrong by -0.000015 seconds (ignored)
+2024-10-02T17:02:59Z chronyd exiting
+----
+
```
[the remark this answers is not in the document]
Comment (citrus-it) [reply docs/debugging-time-sync.adoc:290]: I was structuring this for people who have stuck lab machines/racklettes etc. but that's a good idea. I've added a paragraph to the foot of the document about the post-mortem case.

**Comment by citrus-it** [comment]
> Does chrony have a trace log level where it will log everything about what it's doing? DNS requests, responses (or failures), NTP transactions with remote peers etc. I think we should turn that log up all the way and include it in the output artefacts for both working and failed jobs so we can compare?

We can build chrony with debugging enabled which will provide more detail. That might be something we'd like to do under a variant or similar, or providing a secondary `/usr/sbin/chronyd.debug` for use in post-mortem evidence collection in CI. I'll take a look at this separately. There's also a log file called `selection.log` which is not currently enabled but could be. I'm not sure it would be that useful in the case where we apparently just don't see any available sources - I'm more interested in what adding `-a` to sources reveals there.

**Comment by citrus-it** [comment]
Sorry, squashed and force pushed - bad finger habit from gerrit.

https://github.com/oxidecomputer/omicron/compare/9a054077ea0c5f5c4d4a2b7e7825be8bc135a7c0..6d43d5b4544f5fc8e0bcf2e7a3f92a45cf18182f shows the minor differences from the original set.

**Review by FelixMcFelix — APPROVED** [review]
Things look good to me on the OPTE front, and the added commentary on the new chronyd invocations really help (as someone with little practical experience there!).

The guide is a great addition, thanks for writing that up.

**Pull request merged**